tagged.cabal

```diff
@@ -57,40 +51,22 @@ flag transformers
 library
   default-language: Haskell98
   other-extensions: CPP
-  build-depends: base >= 2 && < 5
+  build-depends:
+    base >= 4.9 && < 5,
+    template-haskell >= 2.11 && < 2.23
   ghc-options: -Wall
   hs-source-dirs: src
-  exposed-modules: Data.Tagged
+  exposed-modules:
+    Data.Proxy.TH
+    Data.Tagged
 
   if impl(ghc >= 9.0)
     -- these flags may abort compilation with GHC-8.10
     -- https://gitlab.haskell.org/ghc/ghc/-/merge_requests/3295
     ghc-options: -Winferred-safe-imports -Wmissing-safe-haskell-mode
 
-  if !impl(hugs)
-    cpp-options: -DLANGUAGE_DeriveDataTypeable
-    other-extensions: DeriveDataTypeable
-
-  if impl(ghc<7.7)
-    hs-source-dirs: old
-    exposed-modules: Data.Proxy
-    other-modules: Paths_tagged
-
-  if impl(ghc>=7.2 && <7.5)
-    build-depends: ghc-prim
-
-  if impl(ghc>=7.6)
-    exposed-modules: Data.Proxy.TH
-    build-depends: template-haskell >= 2.8 && < 2.23
-
   if flag(deepseq)
     build-depends: deepseq >= 1.1 && < 1.6
```

Comment:
`deepseq` was not part of GHC 7.0, but e.g. of 7.4 (and higher).
So maybe the flag could be removed (and deepseq be on always)?

Reply:
The intent of the flag is to allow you to opt out of instances for certain libraries (see also the `transformers` flag, which is in a similar vein), rather than acting as a gate for certain GHC versions. Personally, I am skeptical that such a flag adds much value, but we should have a separate discussion (in a separate issue) if we want to remove it.